frontend/src/store-logic.ts

```diff
@@ -24,11 +24,16 @@ function isUpdateResponse(p: any): p is UpdateResponse {
   return p.type === 'Update'
 }
 
-export function to_app_infos_by_id<T extends { app_id: string }>(app_infos: T[]): Record<string, T> {
-  return app_infos.reduce((acc, appinfo) => {
-    acc[appinfo.app_id] = appinfo
-    return acc
-  }, {} as Record<string, T>)
+function isComplete(elements: Record<string, Partial<AppInfo> | null>): elements is Record<string, AppInfo> {
+  for (const element of Object.values(elements)) {
+    if (element == null) {
+      continue
+    }
+    if (!(!!element.app_id && !!element.date && !!element.description && !!element.name && !!element.size && !!element.source_code_url && !!element.tag_name)) {
```

Comment:
The Bot only returns UpdateResponsese with Partial<AppInfo> as a default. So some sort of Assertion is needed to make Typescript happy and tell it, that the initial request actually has full AppInfos. I think it's better to have a function which might break instead of just using `as` which will fail silently in some cases.
We can also create a new Response type (IniatialMessage) which to send the initial and remove this function. Maybe that is even better.